### ISSUE TITLE
Support ts-ignore, ts-expect-error and ts-nocheck in angular template

### ISSUE THREAD

**Issue by xiaoxiangmoe** [angular/angular#39397]

**Comment by petebacondarwin** [comment]
Can you provide a bit more detail please @xiaoxiangmoe ?
Is the idea that Angular templates could apply these directives to affect the compilation/linting?
How would they be specified - HTML comments?

**Comment by xiaoxiangmoe** [comment]
@petebacondarwin updated.

**Comment by xiaoxiangmoe** [comment]
@petebacondarwin Is there anything else I should do?

**Comment by alxhub** [comment]
This is an interesting idea. I could see this being somewhat useful in the case of migrations, if we wanted to automatically enable something like `strictTemplates` by adding ignore markers for existing broken code.

**Comment by petebacondarwin** [comment]
@alxhub - did you see that we recently enabled this for the Angular eslint plugin?

**Comment by xiaoxiangmoe** [comment]
@alxhub Any idea about this?

**Comment by petebacondarwin** [comment]
This received insufficient votes from the community, which we take as a signal that it may not warrant the effort to implement. but does not mean that it will be closed automatically. We will take the voting into account when deciding whether we should add this to the roadmap or not.

FWIW, I would imagine that this could be implemented simply by capturing these comment nodes in the template, which we already do for angular-eslint, and then write them out almost verbatim, as comments into the generated template typecheck file, so that the TS compiler can act on them...

**Comment by alxhub** [comment]
I agree with @petebacondarwin that this warrants further discussion. What worries me is the lexical nature of such comments - they generally apply to the next line. This is not a concept that the template type-checker understands - line breaks in the TCBs are entirely unrelated to line breaks in the template. We would need some mechanism for adding line breaks to the TCB so that the scope of such comments conformed to the original template - which would not be a trivial feature to support.

**Comment by ippeiukai** [comment]
> What worries me is the lexical nature of such comments - they generally apply to the next line. This is not a concept that the template type-checker understands - line breaks in the TCBs are entirely unrelated to line breaks in the template.

I wonder if the new template syntax could help. Could there be a block syntax for interacting with tools like type-checker for instance?

```html
@compiler ("@ts-expect-error") {
  <app-foo [ffoo]="123">
  </app-foo>
}
```
